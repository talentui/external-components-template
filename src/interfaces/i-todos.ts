// import {IAction} from './i-common'

export enum EFILTER {Active, Finished, All}

export interface ITodoItem {
    contents: string,
    id: string,
    finished: boolean,
}

export interface ITodoItemHanlder {
    onDelete(id: string): void,
    onEdit(id: string, contents: string): void,
    onToggleTodo(id: string): void
}

export interface ITodoHeaderHandler {
    onAdd(contents:string): void
}

export type ITodoHanler = ITodoItemHanlder & ITodoHeaderHandler & ITodoFooterHandler;

export interface ITodoFooterHandler {
    onFilter(filter:EFILTER): void
}

export interface ITodoState {
    filter: EFILTER.All,
    todos:ITodoItem[]
}

export type ITodoItemProps = ITodoItem & ITodoItemHanlder;


export type ITodoListProps = {
    todos: ITodoItem[],
} & ITodoItemHanlder;

export interface IHOFCommonEvent{
    (key: string): (e:React.SyntheticEvent<HTMLElement>) => void;
}

export interface IHOFInputEvent{
    (key: string): (e:React.SyntheticEvent<HTMLInputElement>) => void;
}

export interface ICommonHTMLEvent {
    (e:React.SyntheticEvent<HTMLElement>): void;
}


