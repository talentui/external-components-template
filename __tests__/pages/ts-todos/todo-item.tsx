import * as React from 'react';
import {shallow} from 'enzyme';
import * as renderer from 'react-test-renderer';
import TodoItem from '&/pages/ts-todos/todo-item';
import {ITodoItemProps} from '&/interfaces/i-todos';
  
describe('测试ts todo item', () => {
    let todoEl:React.ReactElement<ITodoItemProps>, todoProps:ITodoItemProps;

    beforeEach(() => {
        todoProps = {
            contents: 'first todo',
            id: '1',
            finished: false,
            onToggleTodo: jest.fn(),
            onDelete: jest.fn(),
            onEdit: jest.fn(),
        }
        todoEl = <TodoItem {...todoProps}/>
    })

    it('快照', () => {
        expect(renderer.create(todoEl)).toMatchSnapshot()
    })

    it('点击todo item, toggle todo 要被调用一次', () => {
        const wrapper  = shallow(todoEl);
        wrapper.simulate('click');
        expect(todoProps.onToggleTodo).toHaveBeenCalledTimes(1)
    })
})