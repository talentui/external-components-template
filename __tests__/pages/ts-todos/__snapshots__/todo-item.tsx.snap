// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试ts todo item 快照 1`] = `
<li
  className="todo-item"
  onClick={[Function]}
>
  <input
    checked={false}
    type="checkbox"
  />
  first todo
  <a
    className="btn-remove"
    href="javascript:;"
    onClick={[Function]}
  >
    xxx
  </a>
</li>
`;
